### DISCUSSION TITLE
Write CSR to las without running readLAS first?

### DISCUSSION THREAD

**Discussion by norrila** [r-lidar/lidR#721] [Q&A] [answered]
HI,

I have a lot of laz files (>1300) that are all missing CRS information and I need to update them all. The only way I was able to "set" the CRS is to read in every file, set CSR and then write a new file.

Is there a way to set the CRS without having to read/write the entire file? The issue is that it takes quite a while to set the projection for all the files.

This is my code, perhaps there is a better way?

```
library(lidR)
library(hutils) # check if output folder exists and makes if not
## ----------------------------------------------------------------------------
# set in/out dir
## ----------------------------------------------------------------------------
in_dir = "D:/temp/las/"
in_las = list.files(in_dir, full.names = TRUE, pattern = ".laz$")
out_dir = paste(in_dir,"_swerfef99TM",.Platform$file.sep, sep="")

# make output folder if it doenst exist
provide.dir(out_dir)

## ----------------------------------------------------------------------------
# set CRS for each file
## ----------------------------------------------------------------------------

for (fname in in_las){
  # read in las, set epsg, 
  las <- readLAS(fname)
  st_crs(las)<-3006 
  
  # write new file
  out_las = gsub(in_dir,out_dir,fname)
  writeLAS(las,out_las)
  }
```

**Comment by Jean-Romain** [comment, marked as answer]
This is not possible. lidR does not aim to replace free and open source parts of lastools. And even with lastools it is not possible with every formats because the VLR must be extended to add the CRS. You can only add the crs without creating a new file if you append the CRS in EVLR i.e. with LAS 1.4 I guess (but I don't know the specs by heart). In both cases, do it with `las2las` it will be much faster than exposing a `data.frame` into R.

**Reply by norrila** [reply]
ok, thanks for the explanation. i didnt realize `las2las `was open source! i thought it was part of the closed source tools.
